components/elm/src/main/elm_driver.F90

```diff
@@ -389,6 +398,8 @@ subroutine elm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate)
 
           call col_ps%Summary(bounds_clump, &
                filter(nc)%num_soilc, filter(nc)%soilc)
+          
+          call BeginGridCBalance(bounds_clump, col_cs, grc_cs)
```

Comment:
Can BeginGridCBalance and BeginGridCBalanceBeforeDynSubgridDriver be combined?  If that's not appealing, perhaps the comment header descriptions should be differentiated.

Reply:
They can't be combined into one as the different data fields are being used in each subroutine for the conservation check

Reply:
I'm not sure I follow.  Both routines have the same arguments `bounds, col_cs, grc_cs`  and they are both just calls to `c2g` with no conservation checks.

Reply:
- In [BeginGridCBalance()](https://github.com/E3SM-Project/E3SM/blob/bishtgautam/lnd/cnp-budget/components/elm/src/biogeochem/EcosystemBalanceCheckMod.F90#L808
), there are 8 fields on which `c2g` is performed (e.g. `totcolc`, `totpftc`, etc)

- In [BeginGridCBalanceBeforeDynSubgridDriver](https://github.com/E3SM-Project/E3SM/blob/bishtgautam/lnd/cnp-budget/components/elm/src/biogeochem/EcosystemBalanceCheckMod.F90#L996), there is only one field (`totcolc`) on which `c2g` is performed.

So, there is only one field, `col_cs%totcolc`, that is common between the two subroutines. But, the `g`-level field in which the output of `c2g` is saved are not the same:
  - `BeginGridCBalance`: uses [`grc_cs%beg_totc`](https://github.com/E3SM-Project/E3SM/blob/bishtgautam/lnd/cnp-budget/components/elm/src/biogeochem/EcosystemBalanceCheckMod.F90#L798)
  - `BeginGridCBalanceBeforeDynSubgridDriver`: uses [`grc_cs%begcb`](https://github.com/E3SM-Project/E3SM/blob/bishtgautam/lnd/cnp-budget/components/elm/src/biogeochem/EcosystemBalanceCheckMod.F90#L993)

At some point in the future, we can revisit to merge these two subroutines when we also have corresponding subroutines for N and P (i.e. `BeginGridNBalance`, `BeginGridPBalance`)

Reply:
All the `c2g` calls output to a different field of `grc_cs` which is why is makes sense to group them together -- especially when they have identical arguments, identical descriptions, and are always called one after the other.  Currently, It appears to be as straightforward as concatenating the two and adding `grc_cs%begcb` to the associate clause of `BeginGridCBalance` (or the other way around).  How will N,P budget developments affect this?

Reply:
Are the `grc_cs` fields used elsewhere even when the budgets are not being computed?  It may make sense to wrap the `BeginGridCBalance` call  in an if statement so that these computations aren't done unless `do_budgets` is true.